Generic type payload and ext access

Ddiidev opened the pull request:
I made the payload generic to make it more practical to use the ext, which I also left public property, to be able to access the field.

The only problem there is the fact that it is not possible to compare a token, e.g. token1 == token2

This is because the compiler has an error in operator overloading, I opened an issue for this

[https://github.com/vlang/v/issues/20777](https://github.com/vlang/v/issues/20777)

Ddiidev committed:
modified payload

Ddiidev committed:
fix ext payload

enghitalo commented:
Tem uma forma melhor the fazer isso aqui tbm. sem precisar usar `.bytes())` você só precisar adicionar o o `encoder` e fazer o encode e pegar o bufer pulando a etapa de transformar o buffer em string

```v
	header := base64.url_encode(json.encode[Header](Header{}).bytes())
	payload_string := base64.url_encode(json.encode[Payload](payload).bytes())
```

Me lembre de por o código aqui

Ddiidev commented:
> Tem uma forma melhor the fazer isso aqui tbm. sem precisar usar `.bytes())` você só precisar adicionar o o `encoder` e fazer o encode e pegar o bufer pulando a etapa de transformar o buffer em string
> 
> ```v
> 	header := base64.url_encode(json.encode[Header](Header{}).bytes())
> 	payload_string := base64.url_encode(json.encode[Payload](payload).bytes())
> ```
> 
> Me lembre de por o código aqui

Manda a boa, joga esse código no peito do pai.

Ddiidev committed:
change json2 to cjson

This change is because json2 doesn't handle complex objects well.
But in this case, a problem is that cjson does not deal with the Time type, so an alias for string is being used

Ddiidev committed:
adjust access of fields

Ddiidev committed:
Refactor time import and JsTime parsing in payload.v

Changed import of 'time' to import the whole module instead of just 'Time'. Updated JsTime's time() method to use 'time.Time' type and improved error handling by using 'return none'. Minor formatting adjustments for consistency.

Ddiidev committed:
Add configurable JWT header and signing options

Ddiidev committed:
Refactor JWT signing/verification via algorithm dispatch

Ddiidev committed:
Merge pull request #1 from Ddiidev/codex/refactor-header-model-to-be-configurable

Make JWT header configurable and add signing options (HS256/RS256)

Ddiidev committed:
Resolve token signing payload segment conflict

Ddiidev committed:
Resolve remaining JWT merge conflicts with payload fallback

Ddiidev committed:
Merge branch 'master' into codex/add-jwt-signing-and-verification-functions

Ddiidev committed:
Merge pull request #2 from Ddiidev/codex/add-jwt-signing-and-verification-functions

Refactor JWT signing/verification via algorithm dispatch

Ddiidev committed:
Add Windows OpenSSL linking and stricter RS256 PEM validation

Ddiidev committed:
Merge pull request #3 from Ddiidev/codex/add-rsa-signing-implementation-in-jwt.v

Add RS256 OpenSSL signer with Windows link flags and stricter PEM validation

Ddiidev committed:
Refine RS256 PEM verification path and expose verifier

Ddiidev committed:
Merge pull request #4 from Ddiidev/codex/add-rs256-signature-verification-function

Refine RS256 verification to rely on OpenSSL PEM parsing

Ddiidev committed:
Refatora opções de credenciais por algoritmo

Ddiidev committed:
Merge pull request #5 from Ddiidev/codex/add-distinct-credential-types-for-signing

Refatora opções de credenciais por algoritmo

Ddiidev committed:
Add RS256 fixture-based JWT tests and testdata docs

Ddiidev committed:
Merge pull request #6 from Ddiidev/codex/add-rs256-tests-and-fixtures

Codex-generated pull request

Ddiidev committed:
docs: expand HS256/RS256 usage and add V CI workflow

Ddiidev committed:
ci: run tests only from project src folder

Ddiidev committed:
fix: make RS256 verification deterministic with valid fixtures

Ddiidev committed:
Merge pull request #7 from Ddiidev/codex/add-rsa256-signing-to-vlang-library

Docs + CI: generic HS256/RS256 README, V CI workflow, and RS256 compatibility fixes
